.github/workflows/tests.yml

```diff
@@ -0,0 +1,40 @@
+name: Tests
+
+on:
+  pull_request:
+    branches: [main, development, feat/*]
+  workflow_dispatch:
+
+jobs:
+  ci:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        php: ['8.1', '8.2', '8.3']
+        dependency-version: [prefer-stable]
+        parallel: ['', '--parallel']
+
+    name: PHP ${{ matrix.php }} - ${{ matrix.os }} - ${{ matrix.dependency-version }} - ${{ matrix.parallel }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools: composer:v2
+          coverage: none
+
+      - name: Setup Problem Matches
+        run: |
+          echo "::add-matcher::${{ runner.tool_cache }}/php.json"
+          echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+      - name: Install PHP dependencies
+        run: composer update --${{ matrix.dependency-version }} --no-interaction --no-progress --ansi
```

Comment:
I would add before this the composer caching with `--prefer-dist`:
```
      - name: Cache dependencies
        uses: actions/cache@v3
        with:
            path: ${{ steps.composer-cache.outputs.dir }}
            key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
            restore-keys: ${{ runner.os }}-composer-
```

Reply:
Good point; I would however add optimizations like this at a later point. Tests run currently in ~40s which should be a good start. It also would need to be considered that we don't have `composer.lock` checked into version control

Reply:
I know, this is not about composer.lock, but about the speed to download the packages from the internet vs having them locally as cache. Seems that given they are downloaded from github itself, this is very fast (2s).